cl/lib/es_signal_processor.py

```diff
@@ -297,12 +307,14 @@ def update_es_documents(
                     throttling_id = get_task_throttling_id(
                         DocketDocument, rel_docket.pk
                     )
-                    update_children_documents_by_query.delay(
-                        es_document.__name__,
-                        rel_docket.pk,
-                        throttling_id,
-                        fields_to_update,
-                        fields_map,
+                    transaction.on_commit(
```

Comment:
I think we could have issues when using `transaction.on_commit` within for loops.

Reading documentation about `transaction.on_commit` it says that it schedules callbacks to the wrapped method to be executed once the transaction is committed.

 For instance, in this case, all the calls to `update_children_documents_by_query` will be scheduled within the for loop to be executed later. However, the problem is that task arguments change within each iteration. So it seems that all the scheduled tasks will be executed once the transaction is committed using the last iteration variable values. 

I did the following test, I created two dockets (with a couple of RDs) assigned to the same Person. Then I changed the Person's name. The result was:

2 `es_document_update` executed that updated the docket for the same last ID.
2 `update_children_documents_by_query` executed that updated the RDs for the same docket ID.

So the first docket and their RDs were not updated.

I wonder if using [functools.partial()](https://docs.python.org/3/library/functools.html#functools.partial) as show in the [docs](https://docs.djangoproject.com/en/4.2/topics/db/transactions/#performing-actions-after-commit) example could help with this issue?

Reply:
Thank you for identifying and bringing attention to this mistake. You're right about the tasks. The closure binds to a name within the outer scope, not the value of that variable at the time the closure is defined. So if the value of that name changes in the outer scope after the definition of the closure, it will change within the closure as well. 

I will use [functools.partial()](https://docs.python.org/3/library/functools.html#functools.partial) to fix this.